### PR TITLE
types: Fix description for year type

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 
1. Fix issue https://github.com/pingcap/tidb/issues/9188, type `year` should be demonstrated with `display width`.

2. Migrate type related tests from TiDB repo to parser repo.

### What is changed and how it works?
add an suffix for type `year` function `FieldType. CompactStr`.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: types/field_type_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
+// Copyright 2015 PingCAP, Inc.
```
Comment (zimulala) [inline types/field_type_test.go]: s/2015/2019

Comment (spongedu) [reply types/field_type_test.go]: Done

Comment (spongedu) [reply types/field_type_test.go]: @zimulala  PTAL :)

**Comment by tiancaiamao** [comment]
PTAL @zimulala

**Comment by zz-jason** [comment]
@spongedu integration test failed, some test case in the TiDB repo should be changed to make it pass.

**Comment by tiancaiamao** [comment]
Wait for this one https://github.com/pingcap/tidb/pull/9253

**Comment by spongedu** [comment]
@zz-jason @tiancaiamao  This pr and https://github.com/pingcap/tidb/pull/9253 seems depend on each other. There seems be a deadlock.....

**Comment by kennytm** [comment]
The proper action, according to the README, is 

1. Merge this PR
2. Update the `parser` dependency in pingcap/tidb#9253
3. Merge pingcap/tidb#9253

**Comment by spongedu** [comment]
@kennytm Agree.  I suppose we should find a way to walk around the test fail.

**Pull request merged**